jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdAuthenticator.java

```diff
@@ -381,17 +401,29 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
         if (uri == null)
             uri = URIUtil.SLASH;
 
-        mandatory |= isJSecurityCheck(uri);
-        if (!mandatory)
-            return new DeferredAuthentication(this);
+        HttpSession session = request.getSession(false);
+        if (_openIdConfiguration.isRespectIdTokenExpiry() && hasExpiredIdToken(session))
+        {
+            logoutWithoutRedirect(request);
+        }
+        else
+        {
+            // If we expired a valid authentication we do not want to defer authentication,
+            // we want to try re-authenticate the user.
```

Comment:
this is kind of confusing as it is in the branch that will be called even if there is not an expired token.
We get here if we don't have an expired token or we are not "respecting" an expired token - i.e ignoring it.
So I'd expect the comment to be about normal processing